### ISSUE TITLE
add model evals into training loop

### ISSUE THREAD

**Issue by rusheb** [AISC-understanding-search/maze-transformer#97]
We want to run the model evals as part of the training loop so that we can measure whether the model performance is increasing. 

Acceptance Criteria:
- model evals are run for every model checkpoint
- evals are logged in wandb (?)

**Comment by rusheb** [comment]
@cmathw  please feel free to add details

**Comment by cmathw** [comment]
Yeah I think at a minimum we would want to include node_overlap and perhaps some custom loss function that @afspies mentioned (logging loss over the predicted path). Is this currently on hold due to the deciding what parts of logging we do on wandb vs keep in the logger functionality and improving path dist eval ( #99 )?

**Comment by valedan** [comment]
I can take this after #99, I've been building up a lot of context on the evals from working on that.

**Comment by mivanit** [comment]
partially completed in #181 

remaining issues:
- no logit evals (implemented logit evals were incorrect, code left in though)
- only the mean of an eval statistic is logged to wandb, and the summary of the statistic is not saved anywhere. I looked into it, and there isnt an easy way to have a plot with error bars on wandb. One way of doing this is just to log more often, and enable smoothing on the dashboard
- evals are not every checkpoint -- `TrainConfig.intervals` lets you set all intervals independently, or `TrainConfig.intervals_count` as a number of times per run
